engine/schema/src/main/java/org/apache/cloudstack/backup/dao/BackupDaoImpl.java

```diff
@@ -142,6 +168,31 @@ public List<Backup> syncBackups(Long zoneId, Long vmId, List<Backup> externalBac
         return listByVmId(zoneId, vmId);
     }
 
+    @Override
+    public Long countBackupsForAccount(long accountId) {
+        SearchCriteria<Long> sc = CountBackupsByAccount.create();
+        sc.setParameters("account", accountId);
+        sc.setParameters("status", Backup.Status.Error, Backup.Status.Failed, Backup.Status.Removed, Backup.Status.Expunged);
+        return customSearch(sc, null).get(0);
+    }
+
+    @Override
+    public Long calculateBackupStorageForAccount(long accountId) {
+        SearchCriteria<SumCount> sc = CalculateBackupStorageByAccount.create();
+        sc.setParameters("account", accountId);
+        sc.setParameters("status", Backup.Status.Error, Backup.Status.Failed, Backup.Status.Removed, Backup.Status.Expunged);
+        return customSearch(sc, null).get(0).sum;
```

Comment:
ensure custom searches ^^^ are not null or empty

Reply:
Same as above.